pkg/fqdn/rulegen.go

```diff
@@ -85,10 +85,6 @@ type RuleGen struct {
 // When config.Cache is nil, the global fqdn.DefaultDNSCache is used.
 func NewRuleGen(config Config) *RuleGen {
 
-	if config.Cache == nil {
```

Comment:
This means we would panic if we con't pass `.Cache` in. We have to either return an error when it is missing or make it a required parameter, perhaps by making it a parameter to `NewRuleGen`.

Reply:
@raybejjani fixed.